### DISCUSSION TITLE
Error while pip installing ifcopenshell

### DISCUSSION THREAD

**Discussion by Vasile-Corjan** [IfcOpenShell/IfcOpenShell#4712] [Q&A] [answered]
Hello everyone!! While trying to install ifcopenshell I got this error. Do you know what the problem can be? pip version is 24.0
![Screenshot 2024-05-21 230339](https://github.com/IfcOpenShell/IfcOpenShell/assets/124814527/a0a22f33-e8f5-4271-90e0-7638f6612cf1)

**Comment by aothms** [comment, marked as answer]
There's a URL in the error report you can visit to get the microsoft C++ build tool, which are required to build binary extension modules on windows. Following the instructions at that link and running the command again should work.

**Reply by Vasile-Corjan** [reply]
Thank you for help!!!

**Comment by maciekspisak** [comment]
I installed recommended build tool but it didn't work. Downloading zip from ifcopenshell page and putting it in site-packages manually instead of using pip worked for me.
It is the problem I don't fully understand but after some research my understanding is that mathutils build work smoothly with python 3.8, not with newer versions.

**Reply by Moult** [reply]
mathutils is now an optional dependency and the plan is to replace it with numpy.
